src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -17,22 +17,30 @@ namespace Microsoft.Docs.Build
 {
     internal class OpsConfigAdapter : IDisposable
     {
-        public static string ValidationServiceEndpoint =>
-            s_isProduction
-            ? "https://docs.microsoft.com"
-            : "https://ppe.docs.microsoft.com";
+        public static string ValidationServiceEndpoint => s_docsEnvironment switch
+        {
+            DocsEnvironment.Prod => "docs.microsoft.com",
+            DocsEnvironment.PPE => "ppe.docs.microsoft.com",
+            DocsEnvironment.Internal => "ppe.docs.microsoft.com",
+            DocsEnvironment.Perf => "ppe.docs.microsoft.com",
+            _ => throw new NotSupportedException()
+        };
 
         public const string BuildConfigApi = "https://ops/buildconfig/";
         private const string MonikerDefinitionApi = "https://ops/monikerDefinition/";
         private const string MetadataSchemaApi = "https://ops/metadataschema/";
         private const string MarkdownValidationRulesApi = "https://ops/markdownvalidationrules/";
 
-        private static readonly string? s_environment = Environment.GetEnvironmentVariable("DOCS_ENVIRONMENT");
-        private static readonly bool s_isProduction = string.IsNullOrEmpty(s_environment) || string.Equals("PROD", s_environment, StringComparison.OrdinalIgnoreCase);
+        private static readonly DocsEnvironment s_docsEnvironment = GetDocsEnvironment();
 
-        private static readonly string s_buildServiceEndpoint = s_isProduction
-            ? "https://op-build-prod.azurewebsites.net"
-            : "https://op-build-sandbox2.azurewebsites.net";
+        private static readonly string s_buildServiceEndpoint = s_docsEnvironment switch
+        {
+            DocsEnvironment.Prod => "https://op-build-prod.azurewebsites.net",
+            DocsEnvironment.PPE => "https://op-build-sandbox2.azurewebsites.net",
+            DocsEnvironment.Internal => "https://op-build-internal.azurewebsites.net",
+            DocsEnvironment.Perf => "https://op-build-internal.azurewebsites.net",
+            _ => "https://op-build-sandbox2.azurewebsites.net",
```

Comment:
thought s_docsEnvironment was default as "PPE" in line 292, and it should still throw here